docs/blog/posts/introduction-to-bpfd.md

```diff
@@ -0,0 +1,605 @@
+blog, bpfd fully supports XDP, TC, Tracepoint, uProbe, and kProbe eBPF programs.
+In addition, bpfd can display information about all types of eBPF programs
+loaded on a system whether they were loaded by bpfd or some other mechanism.
+bpfd is developed in the Rust programming language and uses [Aya], an eBPF
+library which is also developed in Rust.
+
+When used on an individual server, bpfd runs as a system daemon, and
+applications communicate with bpfd using a gRPC API.  bpfd can also be used via
+a command line tool called bpfctl which in turn uses the gRPC API. 
+
+    bpfctl load-from-image -g GLOBAL_u8=01 -i quay.io/bpfd-bytecode/xdp_pass:latest xdp -i bpfd-host -p 100
+
+This architecture is depicted in the following diagram.
+
+![bpfd Architecture](./img/bpfd_architecture.jpg)
+
+Using bpfd in this manner significantly improves security because the API is
+secured using mTLS, and only bpfd needs the privileges required to load and manage
+eBPF programs and maps.
+
+Writing eBPF code is tough enough as it is.  Typically, an eBPF-based
+application would need to also implement support for the lifecycle management of
+the required eBPF programs.  bpfd does that for you and allows you to focus on
+developing your application.
+
+Another key functional advantage that bpfd offers over libbpf or the Cilium
+ebpf-go library is support for multiple XDP programs on a single attach point.
+Standard XDP only supports a single XDP program on a given attach point.
+However, bpfd supports loading multiple XDP programs on a single attach point
+using the [multi-prog] protocol defined in libxdp.  This allows the user to add,
+delete, update, prioritize, and re-prioritize the multiple programs on each
+attach point.  There is also support to configure whether the flow of execution
+should return or continue to the next program in the list based on the return
+value. 
+
+While TC natively supports multiple programs on each attach point, it lacks the
+controls and flexibility enabled by the multi-prog protocol. bpfd therefore also
+supports the same XDP multi-prog solution for TC programs which has the added
+benefit of a consistent user experience for both XDP and TC programs.
+
+eBPF programs are also difficult to debug on a system.  The visibility provided
+by bpfd can be a key tool in understanding what is deployed and how they may
+interact.
+
+[Aya]:https://aya-rs.dev/
+[multi-prog]:https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org
+
+
+### bpfd Kubernetes Support
+
+The benefits of bpfd are brought to Kubernetes by the bpfd operator.  The bpfd
+operator is developed in Go and is based on the Written in Go using the Operator SDK
+framework, so it should be familiar to most Kubernetes application developers.
+The bpfd operator deploys a bpfd-agent pod, containing both bpfd and the bpfd
+agent on each node.  Rather than making requests directly to bpfd with the gRPC
```

Comment:
```suggestion
The bpfd operator deploys a daemonset, containing both the bpfd and bpfd
agent processes.  Rather than making requests directly to bpfd with the gRPC
```